lib/modules/manager/gradle/extract.ts

```diff
@@ -110,11 +110,7 @@ export async function extractAllPackageFiles(
       }
 
       dep.registryUrls = [
-        ...new Set([
-          ...defaultRegistryUrls,
-          ...(dep.registryUrls ?? []),
-          ...registryUrls,
-        ]),
+        ...new Set([...registryUrls, ...(dep.registryUrls ?? [])]),
```

Comment:
Is the ordering change intentional?

Reply:
Yes, it is. `dep.registryUrls` in line 113 is only populated with `plugins.gradle.org` in case the dependency is a Gradle plugin. The ordering change causes any explicitly specified registry URL to be tried before the `plugins.gradle.org` fallback. 

Without this reordering, `plugins.gradle.org` would always be the first-ranked choice, no matter if a private registry is specified or not.